pkg/arch/arch.go

```diff
@@ -41,9 +42,15 @@ func AddSyscallPrefix(symbol string) (string, error) {
 	return addSyscallPrefix(symbol, runtime.GOARCH)
 }
 
-// MustAddSyscallPrefix is like AddSyscallPrefix but panics if the arch is
-// unsupported or if the symbol has already a prefix that doesn't correspond to
-// the running arch.
+// AddSyscallPrefixTestHelper is like AddSyscallPrefix but calls t.Fatal if the
+// arch is unsupported or if the symbol has already a prefix that doesn't
+// correspond to the running arch.
 //
 // It's a helper supposed to be used only in testing code.
-func MustAddSyscallPrefix(symbol string) string
+func AddSyscallPrefixTestHelper(t *testing.T, symbol string) string {
+	syscallName, err := AddSyscallPrefix(symbol)
```

Comment:
but what if we pass a prefix + symbole that corresponds to the right arch?  IMO it should not fail

Reply:
And it will not, it's this case https://github.com/cilium/tetragon/pull/734/files#diff-a79dc57275508064bab63e4361c0a043cd2f4225f0fb8d833448788294cc2ae9R19 (test case here https://github.com/cilium/tetragon/pull/734/files#diff-e7e583e8b6ff00f3e08fabf370b33e954d6419291b7d94f378ce4a165d6663c5L20)

Reply:
nice thanks!